app/services/oauth_clients/connection_manager.rb

```diff
@@ -198,14 +198,12 @@ def get_existing_token
     # Calls client.access_token!
     # Convert the various exceptions into user-friendly error strings.
     def request_new_token(options = {})
-      rack_access_token = rack_oauth_client(options)
-                            .access_token!(:body) # Rack::OAuth2::AccessToken
+      rack_access_token = rack_oauth_client(options).access_token!(:body)
 
       ServiceResult.success(result: rack_access_token)
-    rescue Rack::OAuth2::Client::Error => e # Handle Rack::OAuth2 specific errors
+    rescue Rack::OAuth2::Client::Error, Faraday::ParsingError => e
       service_result_with_error(i18n_rack_oauth2_error_message(e), e.message)
-    rescue Timeout::Error, EOFError, Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, Net::ProtocolError,
-           Errno::EINVAL, Errno::ENETUNREACH, Errno::ECONNRESET, Errno::ECONNREFUSED, JSON::ParserError => e
+    rescue Faraday::TimeoutError, Faraday::ConnectionFailed => e
```

Comment:
Add `Faraday::SSLError` due to the possibility for it to be raised by Faraday in case of ssl specific issues.